src/runtime/BorrowedReference.cs

```diff
@@ -0,0 +1,26 @@
+namespace Python.Runtime
+{
+    using System;
+    readonly ref struct BorrowedReference
+    {
+        readonly IntPtr pointer;
+        public bool IsNull => this.pointer == IntPtr.Zero;
+
+        public PyObject ToPyObject()
+        {
+            if (this.IsNull) throw new NullReferenceException();
+
+            Runtime.XIncref(this.pointer);
+            return new PyObject(this.pointer);
+        }
+
+        [Obsolete("Use overloads, that take BorrowedReference or NewReference")]
+        public IntPtr DangerousGetAddress()
```

Comment:
Use a property named `Address` can be more fit?
Why I have to be warned if I trying use the raw pointer, seems it doesn't make any sense. Or just don't expose the interface and use `explicit operator IntPtr` instead.

Reply:
This is modeled after `SafeHandle` class and its `DangerousGetHandle` method.
The whole point of introducing these references is to get rid of `IntPtr`s entirely. It should warn you against converting into `IntPtr`. Maybe `Obsolete` part is unnecessary as `Dangerous` sounds like a warning enough. I will remove `Obsolete`.